Applications/MNIST/README.md

```diff
@@ -0,0 +1,141 @@
+...
+
+```
+
+In order to get the training data, ```DataBufferFromCallback``` option is used and ```getData_train``` and ```getData_val``` functions are defined which reads the feature and label with batch_size and fill the databuffer. If there is no data for an epoch, then ```last``` will be true.
+
+``` c++
+/**
+ * @brief      get data which size is batch for train
+ * @param[out] outVec
+ * @param[out] outLabel
+ * @param[out] last if the data is finished
+ * @param[in] user_data private data for the callback
+ * @retval status for handling error
+ */
+int getBatch_train(float **outVec, float **outLabel, bool *last,
+                   void *user_data);
+```
+
+### MNIST Configuration file
+The model configuration is in ```res/mnist.ini```. ADAM optimizer is used and the trained model is going to be saved in ```model.bin```.
+
+``` bash
+# Model Section : Model
+[Model]
+Type = NeuralNetwork	    # Network Type NeuralNetwork
+Learning_rate = 1e-4 	    # Learning Rate
+Epochs = 1500		        # Epochs
+Optimizer = adam 	        # Optimizer : adam (Adamtive Moment Estimation)
+Loss = cross  		        # Loss function : cross ( for cross entropy )
+Save_Path = "model.bin"  	# model path to save / read
+batch_size = 32		        # batch size
+beta1 = 0.9 		        # beta 1 for adam
+beta2 = 0.999	            # beta 2 for adam
+epsilon = 1e-7	            # epsilon for adam
+
+# Layer Section : Name
+[inputlayer]
+Type = input
+Input_Shape = 1:28:28
+
+# Layer Section : Name
+[conv2d_c1_layer]
+Type = conv2d
+kernel_size = 5,5
+bias_initializer=zeros
+Activation=sigmoid
+weight_initializer = xavier_uniform
+filters = 6
+stride = 1,1
+padding = 0,0
+
+[pooling2d_p1]
+Type=pooling2d
+pool_size = 2,2
+stride =2,2
+padding = 0,0
+pooling = average
+
+[conv2d_c2_layer]
+Type = conv2d
+kernel_size = 5,5
+bias_initializer=zeros
+Activation=sigmoid
+weight_initializer = xavier_uniform
+filters = 12
+stride = 1,1
+padding = 0,0
+
+[pooling2d_p2]
+Type=pooling2d
+pool_size = 2,2
+stride =2,2
+padding = 0,0
+pooling = average
+
+[flatten]
+Type=flatten
+
+[outputlayer]
+Type = fully_connected
+Unit = 10
+weight_initializer = xavier_uniform
+bias_initializer = zeros
+Activation = softmax
+
+```
+
+### How to run
+Once you compile the codes, you can run with
+
+``` bash
+$ ./nntrainer_mnist ${res}/mnist.ini
+```
+
+For the comparison, we provide Tensorflow code for same model and dataset in ```Tensorflow/Training_Keras.py```.
+
+```dataset.py``` is for the data generation from the input data file, ```mnist_trainingSet.dat```.
+
+
+You can run with
+
+``` bash
+$ python3 Training_Keras.py
+```
+
+```mnist_trainingSet.data``` must be in the same directory with ```Training_Kears.py```.
+
+
+### Comparison with Tensorflow
+This is the comparison with tensorflow-1.14.0 for the two cases. One is with zero initialization of weight and bias and the other is weight initialization data from tensorflow. As you can see, the results are pretty much same.
```

Comment:
the other is weight initialization data from tensorflow -> the other is `random` weight initialization data `using the default intializers for each layer` from tensorflow

Reply:
done.